share/eds/store.go

```diff
@@ -419,3 +431,30 @@ func setupPath(basepath string) error {
 	}
 	return nil
 }
+
+// inMemoryOnceMount is used to allow reading once from buffer before using main mount.Reader
+type inMemoryOnceMount struct {
+	buf []byte
```

Comment:
Why not embed the bytes.Buffer? Then inMemoryReader won't be needed, nor the buffer on 189.

The code would look like:
WriteEDS(mount)

mount.WriteTo(file)

RegisterShard(mount)

Reply:
Thank for suggested, I've made small refactoring to incorporate buffer into mount. 
inMemoryReader is needed to extend bufio.Reader to be compatible with mount.Mount interface.